lib/internal/process/warning.js

```diff
@@ -90,7 +95,12 @@ function doEmitWarning(warning) {
 }
 
 function onWarning(warning) {
+  const disableWarnings = getOptionValue('--disable-warnings');
```

Comment:
Should the first use create a `SafeSet` from this array and then we check if the set has the emitted warning?

Reply:
Done 👍